src/components/cc-icon/cc-icon.js

```diff
@@ -0,0 +1,109 @@
+import { css, html, LitElement } from 'lit';
+import { unsafeSVG } from 'lit/directives/unsafe-svg.js';
+
+/**
+ * @typedef {import('./cc-icon.types.js').IconModel} IconModel
+ * @typedef {import('./cc-icon.types.js').IconSize} IconSize
+ */
+
+/**
+ * A component rendering an SVG icon.
+ *
```

Comment:
I think we could introduce an `Accessibility guidelines` section because `<svg>` a11y is a tricky subject and we want to offer some guidance on how to use the `accessible-name` prop.

![image](https://user-images.githubusercontent.com/100240294/205882990-b271b833-5df4-413a-9e77-f603e7a61731.png)


```suggestion
 * **Accessibility guidelines:**
 *
 * Use the `accessible-name` prop only if your icon provides information that is not already given in its surrounding text.
 *
 * If this prop has a value:
 *
 *  * sets a `role="img"` and `aria-label="the value of this prop"` attributes on the `<svg>` element.
 *  * creates a `<title>` element inside the `<svg>`.
 *
 * This allows assistive technologies to recognize the element as an image and convey its information.
 *
 * If this prop has no value, sets an `aria-hidden="true"` attribute on the `<svg>` element so that it can be ignored by assistive technologies.
```

You'll see below that I also suggest we add this info in the `accessible-name` prop doc.It might seem redundant but I think both the main doc and the prop doc should mention this because if you don't know about accessibility, you might not even check the `accessible-name` prop :roll_eyes:.

Feel free to rephrase any of these of course :wink: